jinahub/encoders/text/DPRTextEncoder/dpr_text.py

```diff
@@ -0,0 +1,151 @@
+        encoder_type: Literal['context', 'question'] = 'question',
+        base_tokenizer_model: Optional[str] = None,
+        title_tag_key: Optional[str] = None,
+        max_length: Optional[int] = None,
+        default_batch_size: int = 32,
+        default_traversal_paths: List[str] = ['r'],
+        device: str = 'cpu',
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        self.device = device
+        self.max_length = max_length
+        self.title_tag_key = title_tag_key
+
+        if encoder_type not in ['context', 'question']:
+            raise ValueError(
+                'The ``encoder_type`` parameter should be either "context"'
+                f' or "question", but got {encoder_type}'
+            )
+        self.encoder_type = encoder_type
+
+        if not base_tokenizer_model:
+            base_tokenizer_model = pretrained_model_name_or_path
+
+        if encoder_type == 'context':
+            if not self.title_tag_key:
+                warnings.warn(
+                    'The `title_tag_key` argument is not set - it is recommended'
+                    ' to encode the context text together with the title to match the'
+                    ' model pre-training. '
+                )
+            self.tokenizer = DPRContextEncoderTokenizerFast.from_pretrained(
+                base_tokenizer_model
+            )
+            self.model = DPRContextEncoder.from_pretrained(
+                pretrained_model_name_or_path
+            )
+        elif encoder_type == 'question':
+            self.tokenizer = DPRQuestionEncoderTokenizerFast.from_pretrained(
+                base_tokenizer_model
+            )
+            self.model = DPRQuestionEncoder.from_pretrained(
+                pretrained_model_name_or_path
+            )
+
+        self.model = self.model.to(torch.device(self.device)).eval()
+
+        self.default_traversal_paths = default_traversal_paths
+        self.default_batch_size = default_batch_size
+
+    @requests
+    def encode(self, docs: Optional[DocumentArray], parameters: dict, **kwargs):
+        """
+        Encode all docs with text and store the encodings in the embedding
+        attribute of the docs.
+
+        :param docs: documents sent to the encoder. The docs must have the
+            ``text`` attribute.
+        :param parameters: dictionary to define the ``traversal_path`` and the
+            ``batch_size``. For example,
+            ``parameters={'traversal_paths': ['r'], 'batch_size': 10}``
+        """
+
+        if docs:
+            document_batches_generator = get_docs_batch_generator(
+                docs,
+                traversal_path=parameters.get(
+                    'traversal_paths', self.default_traversal_paths
+                ),
+                batch_size=parameters.get('batch_size', self.default_batch_size),
+                needs_attr='text',
+            )
+
+            for batch_docs in document_batches_generator:
```

Comment:
maybe this and below can be refactored into a private method. As it is, this method is too large.

Reply:
Why do you think it is too large? It contains very little function/method calls, mostly it's just line breaking of single statements

Reply:
not a big deal. I just prefer it when it's broken up, like in here, for example https://github.com/jina-ai/executors/blob/4d260e2206c1e214369a704cd17324b6aa822ca5/jinahub/encoders/image/CLIPImageEncoder/clip_image.py#L68

as you wish